### DISCUSSION TITLE
Is it possible to statically compile python code to WASM?

### DISCUSSION THREAD

**Discussion by Duncanr-glitch** [pyodide/pyodide#4046] [Q&A]
Is there any way to take my python files and statically compile them down to WASM files for my JavaScript using pyodide?

**Comment by ryanking13** [comment]
There are some projects that try to compile Python (or Python-like) codes into a binary.

DISCLAIMER: None of these are used in Pyodide, and I haven't used these before. But you might be interested :)

- [mypyc](https://github.com/mypyc/mypyc): compiled type-annotated codes into a C-extension. It may be possible to compile to WASM?
- [spy](https://github.com/spylang/spy): @antocuni introduced this in Pycon US.
- [lpython](https://github.com/lcompilers/lpython): It is introduced about a week ago.
